plugins/TwoFactorAuth/Controller.php

```diff
@@ -287,6 +312,14 @@ public function showRecoveryCodes()
             throw new Exception('You have to verify your password first.');
         }
 
+        if (!$regenerateSuccess && !$regenerateError) {
```

Comment:
Looking more, `$regenerateError` is set below this, so I guess either it shouldn't be in this if, or this if should be below the next one?

Reply:
yeah, the logic is funny here, because if the `$regenerateSuccess` is false doesn't mean it was an error, we use this method when we show the codes, even when we don't regenerate.
but the `$regenerateError` part is true, probably have to move it after it. I recheck how this method works.